### PR TITLE
Update vendor of buildah and containers/images

### PR DESCRIPTION
**Pull request by rhatdan**
Mainly add support for podman build using --overlay mounts.

Updates containers/image also adds better support for new registries.conf
file.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon @baude @vrothberg @giuseppe @TomSweeneyRedHat PTAL

**Comment by mheon** [comment]
LGTM, but I don't know enough about the login code to say what's changed - would like someone else to look that over

**Comment by rhatdan** [comment]
@mheon This is the same change we made to buildah login.  The interface disappeared since GetAuthentiation returns the same data.

File: docs/podman-build.1.md [hunk 1]
```diff
@@ -519,6 +521,21 @@ content label. Shared volume labels allow all containers to read/write content.
...
+
+  One use case of the `overlay` mount is sharing the package cache from the host into the container to allow speeding up builds.
+
+  Note:
+
+     - Overlay mounts are not currently supported in rootless mode.
+     - The `O` flag is not allowed to be specified with the `Z` or `z` flags. Content mounted into the container is labeled with the private label.
+       On SELinux systems, labels in the source directory needs to be readable by the container label. If not, SELinux container separation must be disabled for the container to work.
+     - Modification of the directory volume mounted into the container with an overlay mount can cause unexpected failures.  It is recommended that you do not modify the directory until the container finishes running.
+
```
Comment (TomSweeneyRedHat) [inline docs/podman-build.1.md]: :+1:

**Comment by rhatdan** [comment]
@rh-atomic-bot r=TomSweeneyRedHat

**Comment by mheon** [comment]
Oh dear - does Homu even work on this repo anymore?
/lgtm

**Comment by baude** [comment]
@rhatdan not sure what to do here but maybe repush?

**Comment by mheon** [comment]
Green button pushed

**Pull request merged**